### PR TITLE
Add OptionalGetElement Experimental Api

### LINKED ISSUE #21591: OptionalGetElement

### PR DESCRIPTION
**Pull request by root-yash**
Close #21591

### CODE REVIEW

**Comment by root-yash** [comment]
Close #21591

**Comment by root-yash** [comment]
> Hey @root-yash, I guess you missing the function implementation for our backends, you can find info about backend APIs in our [deep dive](https://unify.ai/docs/ivy/overview/deep_dive/navigating_the_code.html#backend-api). Also refer to our [ivy experimental API open task](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#ivy-experimental-api).
> 
> For now sadly I can't merge your PR unless you have the function implemented in all backends Ivy support.

Hey @KareemMAX, 
Thank you for your input. I had implemented a [Standalone function](https://unify.ai/docs/ivy/overview/deep_dive/function_types.html#standalone-functions) as there is no specific backend implementation that is needed. Taking this PR as a reference https://github.com/unifyai/ivy/pull/14705/files. Should I implement it for all backends?

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            "uint16",
+            "uint32",
+        ],
+        min_value=-100,
+        max_value=100,
+        min_num_dims=0,
+        max_num_dims=5,
+        min_dim_size=0,
+        max_dim_size=5,
+        allow_nan=True,
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py:31]: I would suggest if you removed all of those constraints. As the input can be any tensor type, or list of tensors.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+):
+    input_dtype, x = dtype_and_x
+
+    helpers.test_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        on_device=on_device,
+        backend_to_test=backend_fw,
+        fn_name=fn_name,
+        x=x[0],
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py:50]: You would need to test if the input is a list or a tensor. I would suggest if you drawn a `bool` which decides if you should input a list or a tensor.
```suggestion
        x=x[0] if input_tensor else x,
```

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @root-yash,
Really sorry for the misunderstanding, I got confused when I didn't see any backend implementations. My bad 😅.

You code looks good to me. I just have some comments on your test. Can you please resolve them?

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py (hunk 2 above)
Comment (root-yash) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py:50]: > You would need to test if the input is a list or a tensor.

Sorry, I am not able to understand what you mean. Can you please explain?
And below is the output of what x is assigned to:
```
[array(0, dtype=int8)]
[array(-2.44758124e-308-2.44758124e-308j)]
[array([-3.33333333e-001-9.00719925e+015j,
        4.61104000e+016-2.22044605e-016j,
        1.40129846e-045+1.10000000e+000j,
       -3.62740835e-126-5.00000000e-001j,
       -1.00000000e+000+8.85831200e+281j,
        2.34160834e+016+1.00000000e-005j,
        3.40282347e+038-9.00719925e+015j,
        6.10351562e-005-4.78016237e+016j])]
[array(False)]
```

**Comment by KareemMAX** [comment]
I've modified your test to support lists as instructed by ONNX. But for some reason that made your code fail tests. I believe this issue is not caused by you, you don't have to do anything now. While I'll work with the team to see what is causing this.

**Comment by KareemMAX** [comment]
Hey @root-yash,
I'm really sorry for the delay. We are still investigating the testing issue, will try to fix that soon. Sorry for keeping you waiting

**Comment by root-yash** [comment]
> Hey @root-yash,
> I'm really sorry for the delay. We are still investigating the testing issue, will try to fix that soon. Sorry for keeping you waiting

Thanks for letting me know :)

**Review by KareemMAX — APPROVED** [review]
Hey @root-yash,
I've talked with @sherry30 and as I think the issue is with our testing pipeline instead of your code. I'll merge your code now. Really sorry for the huge delay 😅

**Pull request merged**